src/features/room/helpers/preselect.tsx

```diff
@@ -56,174 +55,154 @@ export default function preselect({
+                        yield toast.open({
+                            type: ToastType.Error,
+                            title: i18n('preselectToast.unauthorizedTitle'),
+                        })
 
-                // Only pin public rooms.
-                const pinned = !permissions.length
+                        return
+                    }
 
-                const numModded: number = yield db.rooms
-                    .where({ owner, id: roomId })
-                    .modify({ pinned, hidden: false })
+                    // Only pin public rooms.
+                    const pinned = !permissions.length
 
-                if (numModded !== 0) {
-                    return
-                }
+                    const numModded: number = yield db.rooms
+                        .where({ owner, id: roomId })
+                        .modify({ pinned, hidden: false })
 
-                try {
-                    yield db.rooms.add({
-                        createdAt,
-                        createdBy,
-                        id: stream.id,
-                        name,
-                        owner,
-                        pinned,
-                    })
-                } catch (e: any) {
-                    // Account for race conditions…
-                    if (!/uniqueness/.test(e?.message || '')) {
-                        throw e
+                    if (numModded !== 0) {
+                        return
                     }
-                }
 
-                yield selectRoom(roomId, owner)
-            } catch (e) {
-                dismissToast = false
+                    try {
+                        yield db.rooms.add({
+                            createdAt,
+                            createdBy,
+                            id: stream.id,
+                            name,
+                            owner,
+                            pinned,
+                        })
+                    } catch (e: any) {
+                        // Account for race conditions…
+                        if (!/uniqueness/.test(e?.message || '')) {
+                            throw e
+                        }
+                    }
 
-                tc = yield retoast(tc, {
-                    type: ToastType.Error,
-                    title: i18n('preselectToast.failedtoOpenTitle'),
-                })
-            }
+                    yield selectRoom(roomId, owner)
+                } catch (e) {
+                    yield toast.open({
+                        type: ToastType.Error,
+                        title: i18n('preselectToast.failedtoOpenTitle'),
+                    })
+                }
+            })
         } finally {
-            if (dismissToast) {
-                setTimeout(() => {
-                    tc?.dismiss()
-                }, 2000)
-
-                yield delay(2000)
-            }
+            yield toast.dismiss({
+                asap: yield cancelled(),
+            })
```

Comment:
The new `dismiss` takes care of the delay in more elegant way. 

The problem we're fixing here is that very often rooms open quickly. In fact so quickly that the toast pops up and slides away almost instantly leaving bad aftertaste of glitchy UI. Toasts stay up for at least 1 second.